sdk/eventhub/event-hubs/src/models/public.ts

```diff
@@ -178,6 +178,81 @@ export interface EventHubConsumerClientOptions extends EventHubClientOptions {
   loadBalancingOptions?: LoadBalancingOptions;
 }
 
+/**
+ * Describes the options that can be provided while creating the EventHubProducerClient.
+ * - `enableIdempotentPartitions`: Indicates whether or not the EventHubProducerClient should enable idempotent publishing to Event Hub partitions.
+ * - `userAgent`        : A string to append to the built in user agent string that is passed as a connection property
+ * to the service.
+ * - `webSocketOptions` : Options to configure the channelling of the AMQP connection over Web Sockets.
+ *    - `websocket`     : The WebSocket constructor used to create an AMQP connection if you choose to make the connection
+ * over a WebSocket.
+ *    - `webSocketConstructorOptions` : Options to pass to the Websocket constructor when you choose to make the connection
+ * over a WebSocket.
+ * - `retryOptions`     : The retry options for all the operations on the client/producer/consumer.
+ *    - `maxRetries` : The number of times the operation can be retried in case of a retryable error.
+ *    - `maxRetryDelayInMs`: The maximum delay between retries. Applicable only when performing exponential retries.
+ *    - `mode`: Which retry mode to apply, specified by the `RetryMode` enum. Options are `Exponential` and `Fixed`. Defaults to `Fixed`.
+ *    - `retryDelayInMs`: Amount of time to wait in milliseconds before making the next attempt. When `mode` is set to `Exponential`,
+ *       this is used to compute the exponentially increasing delays between retries. Default: 30000 milliseconds.
+ *    - `timeoutInMs`: Amount of time in milliseconds to wait before the operation times out. This will trigger a retry if there are any
+ *       retry attempts remaining. Minimum value: 60000 milliseconds.
+ *
+ * A simple usage can be `{ "maxRetries": 4 }`.
+ *
+ * Example usage:
+ * ```js
+ * {
+ *     retryOptions: {
+ *         maxRetries: 4
+ *     }
+ * }
+ * ```
+ */
+export interface EventHubProducerClientOptions extends EventHubClientOptions {
+  /**
+   * Indicates whether or not the EventHubProducerClient should enable idempotent publishing to Event Hub partitions.
+   * If enabled, the producer will only be able to publish directly to partitions;
+   * it will not be able to publish to the Event Hubs gateway for automatic partition routing
+   * nor will it be able to use a partition key.
+   * Default: false
+   */
+  enableIdempotentPartitions?: boolean;
+  /**
+   * The set of options that can be specified to influence publishing behavior,
+   * specified per partition.
+   *
+   * These options are intended to be used for specialized scenarios,
+   * such as when recovering the state used with idempotent publishing.
+   *
+   * It is highly recommended that these options only be specified if there
+   * is a proven need to do so.
```

Comment:
In this case, I disagree with Richard's feedback strongly.  Unless I'm mistaken, Richard is referring to the feature itself - which you have documented for `enableIdempotentPartitions` in a much more approachable way.   This set of `partitionOptions`  isn't something that 95% of the users of the idempotent feature should use.   If you don't know that you need to use it, you probably shouldn't be using it.

Granted, I intentionally left it somewhat generic so that we could add options specific to a partition that are not bound to idempotent publishing in the future, and that would be the point that I'd reconsider toning down the language.